src/pages/docs/how-to/http-bridge.mdx

```diff
@@ -0,0 +1,676 @@
+          "with": "did:key:z6Mkabc123"
+        }
+      },
+      "prf": [],
+      "ran": {
+        "/": "bafyreiclycv7vdfutbtakytkbvcpx2ontlp3fusfb7niugvy5a6kawnhk4"
+      }
+    },
+    "s": {
+      "/": {
+        "bytes": "7aEDQDosaCizSSLXXm55dAeuDACNw2H9/JoUa6oLkLY7mv6TXeW+Z0xJouYj1MGf6LpHn756IW2NprqXq4c20EjlOwM"
+      }
+    }
+  },
+  {
+    "p": {
+      "fx": {
+        "fork": []
+      },
+      "iss": "did:web:web3.storage",
+      "meta": {},
+      "out": {
+        "ok": {
+          "root": {
+            "/": "bafybeidhkumeonuwkebh2i4fc7o7lguehauradvlk57gzake6ggjsy372a"
+          },
+          "shards": [
+            {
+              "/": "bagbaiera4tntawdwlxf33uld7hd5yfaadct5galsr45vawbomjbiytdx4dzq"
+            },
+            {
+              "/": "bagbaieraw4kz73gtuttgxlckdby7fezy4o5hdfspcrl7nqxsh3ultzse2q2a"
+            }
+          ]
+        }
+      },
+      "prf": [],
+      "ran": {
+        "/": "bafyreihobt6uxochgcmfs5bhetynrccrqsuipwp5cy5xxclvqgzw7qsmui"
+      }
+    },
+    "s": {
+      "/": {
+        "bytes": "7aEDQIa1JFOzEWHHfUZQJqRhZzElISfhq4b9Xqca0TgjX+3N7SLrYTB6SN/2GdLSgkRuIKZivB+tSdG4s/+Fvh1CxAQ"
+      }
+    }
+  }
+]
+```
+
+Remember that if the `status` field of the `store/add` result is `upload` you will then need to `PUT` your file to the
+to the given URL with the given headers.
+
+## Sharded Uploads
+
+If you'd like to store a file larger than 5GB you'll need to shard it into smaller CARs before storing. We don't
+proscribe any particular sharding algorithm - you'll need to:
+
+1. read the input CAR
+2. write the first CAR file header
+3. write blocks until the first CAR hits your limit (a number lower than 5GB)
+4. write a new CAR file header for the second CAR
+5. write blocks until the second CAR hits your limit
+6. repeat until end of blocks
+
+If you're implementing this yourself it may be helpful to look
+into [our JavaScript implementation](https://github.com/web3-storage/w3up/blob/d6978d7ab299be76987c6533d18e6857f6998fe6/packages/upload-client/src/index.js#L38).
+We provide an example here using the [`carbites-cli`](https://www.npmjs.com/package/carbites-cli) utility:
+
+First we'll create an absolutely gigantic file:
+
+```sh
+echo "this is an absolutely gigantic file created by travis" > big.txt
```

Comment:
I'd use a bigger file for this example. Since the target split size is so small, the size of the CARs that were output looks wrong (since the eventual size includes the encoded CAR header, as well as the per block headers).

Sorry it was the simpliest thing that could work...the requested `size` is a "target" size and only counts block bytes.

It also looks like there's a bug where if you run out of blocks at the threshold you'll get an empty CAR 🤦 . Like, I'd expect 2 CARs here, not 3 - 1 CAR containing the directory block and 1 containing the file block.

Reply:
great idea - done and done!